src/serac/physics/tests/serac_solid_functional_shape.cpp

```diff
@@ -161,10 +160,15 @@ TEST(solidFunctionalShape, manualNodes)
 
   for (int i = 0; i < shape_displacement.Size(); ++i) {
     // Check the final displacement norm
-    EXPECT_NEAR(shape_displacement[i], pure_displacement[i], 1.0e-11);
+
+    // TODO why is this tolerance so high? It should be closer to machine precision.
```

Comment:
We should be testing relative errors, it's hard to understand if `2.0e-10` is a significant absolute error or not without knowing the characteristic dimensions of the problem

Reply:
Agree with relative. Though in this case, I doubt the displacements are large enough to explain the problem - looking at the loads and the elastic constants, I don't think the displacements are $~10^5$. I'm pretty sure the problem is the nonlinear solver tolerance.

Reply:
If the linearized kinematics case also needs the large(ish) tolerances, I don't know what the heck the problem is.

Reply:
This error also exists in the linear case. My best guess is the projection I'm using to the mesh nodes is not the same as simple addition. I'll investigate this more tomorrow.

Reply:
> Though in this case, I doubt the displacements are large enough to explain the problem

My point was that the displacements in this problem are small, so the relative errors would be even larger than the absolute errors (but not clear how much bigger)

Reply:
Also: we don't need to be doing componentwise relative error calculations, we can just do 
$$\frac{||\boldsymbol{u}_s - \boldsymbol{u}||}{||\boldsymbol{u}||}$$

and not have to worry about division by zero per component.

Reply:
After tightening the tolerances,
```cpp
  SolverOptions options = {DirectSolverOptions{.print_level = 0}, solid_mechanics::default_nonlinear_options};

  options.nonlinear.abs_tol = 1.0e-14;
  options.nonlinear.rel_tol = 1.0e-14;
```
the relative errors for the linear and nonlinear cases are

```cpp
  double error = pure_displacement.DistanceTo(shape_displacement.GetData());
  double relative_error = error / pure_displacement.Norml2();
  std::cout << relative_error << std::endl;
```

4.10296e-16 and 2.04338e-15, respectively.